# What's in a Pangenome {#pv}

> "That which is striking and beautiful is not always good, but that which is good is always beautiful."
>
> --- Anne "Ninon" de l'Enclose

<span class="newthought">This chapter will introduce the interactive</span> pangenome visualization <span class="abr">PanViz</span>. The visualization is an attempt to challenge the current approaches to mapping pangenome data to visual elements, by providing a visualization of the distribution of gene groups in a way that scales well with the number of genomes and gene groups. Scalability in data visualization can effectively be achieved by defining meaningful summary statistics for the data points the visualization should scale with. In the case of <span class="abr">PanViz</span>, and pangenome data in general, the two dimensions that are subject to scale change are the number of genomes and the number of gene groups. Visualizations of synteny (see section \@ref{showing-genome-relationships}) often scales with the number of genomes and quickly reach a maximum. Contrary to this, <span class="abr">PanViz</span> uses gene groups as the primary data. In order to provide a scalable visualization of gene groups, <span class="abr">PanViz</span> relies on two natural classification schemes. The first is the core/accessory/singleton division indigenous to pangenome analysis, the second is classification into the top level biological processes provide by the *Gene Ontology* [@Ashburner:2000ja]. By visualizing the levels of these two classification schemes and scale the size of each level to the number of gene groups they represent, <span class="abr">PanViz</span> effectively makes the number of visual element independent on the number of gene groups. The two classification schemes are fundamentally different in nature: the classification by biological process describes an inherent quality of the gene group itself, while the pangenome classification describes a relationship between the genomes in the pangenome. This difference means that the gene ontology classification is static while the pangenome classification is subject to change if a different selection of genomes is investigated, e.g. by subsetting the pangenome. This duality in classification leads to the main interaction implemented in <span class="abr">PanViz</span> where the user can select different subsets of the pangenome and follow how gene groups from different gene ontology classifications change their pangenome classification. The utility of this is in detecting structure in how different biological processes are related to the current selection of genomes. In <span class="abr">PanViz</span> the lowest number of genomes in a pangenome subset is two, though it could be argued that a single genome could be viewed as a pangenome. In order to look at the genome level it is possible to compare up to two different genomes with the currently selected pangenome. This visualization shows how the content of each genome is represented in the pangenome, e.g. if most of the genes related to metabolic processes are part of the core or, if the selected genome is not part of the pangenome, whether certain biological processes are less represented than others. The choice of only looking at the top level biological processes is necessary in order to limit the number of visual elements in the visualization. This choice impose a lack of granularity though, as e.g. metabolic processes covers a wide variety of more detailed terms within the gene ontology. In order to investigate the gene ontology classification of the gene groups in more detail, a third visualization is provided. This visualization maps the sub-terms of the gene ontology onto a zoom-able treemap weighted by the number of gene groups each sub-term covers. In addition to the pure visualization part it is also possible to extract raw gene group information from <span class="abr">PanViz</span>. This is implemented as a visual querying system, relying on the fact that all visual elements in <span class="abr">PanViz</span> represent a set of gene groups. The gene groups represented by an element can be selected and subsequently refined by applying set operations (intersection, union, etc.) between the current selection and other visual elements.

The following article, submitted to [Bioinformatics](http://bioinformatics.oxfordjournals.org), gives a brief overview of the <span class="abr">PanViz</span> visualization. Figure <span class="abr">S</span>1 from the supplementary material is added to the end to aid the reader. The supplementary figure is available in appendix \ref{PVsupfig} and the an example of the visualization is available in appendix \ref{supdata}.

**Article 2**

[PanViz: Interactive Visualization of the Structure of Functionally Annotated Pangenomes](articles/panviz_appl_note.pdf)

The goal with <span class="abr">PanViz</span> was to create a rich, scalable, and interactive visualization of gene group distributions. Some of the challenges facing such a utility when it comes to gaining wide adoption is in how easily it can fit into the work-flow of researchers. Often interactive visualizations are tied to specific software, making it potentially difficult to incorporate into existing pipelines. Further, reliance on a specific software makes these visualizations more difficult to share with collaborators stymieing their use as communicative tools. <span class="abr">PanViz</span> handles these challenges by being implemented as a completely self-contained <span class="abr">HTML</span> file that can be opened in all modern browsers. Both the <span class="abr">JavaScript</span> code powering the visualization as well as the data is embedded within the file resulting in a visualization that can be accessed by anyone with a reasonable modern computer. This solves the issue of share-ability, but generating the visualization still requires an easily accessible interface. <span class="abr">PanViz</span> solves this by delegating the creation of new visualizations to the <span class="abr">PanVizGenerator</span> <span class="abr">R</span> package. This package makes it easy to provide native support for different tools, such as <span class="abr">FindMyFriends</span>, as well as provide generic functions that can be accessed by unsupported tools. Whether the choices taken with regards to the implementation will result in a tool that is easier to adopt and share, thus leading to higher usage has yet to be seen as the tool is still too new assess adoption.
